core/src/main/java/com/linecorp/armeria/common/util/TransportType.java

```diff
@@ -96,7 +97,10 @@ public Class<? extends SocketChannel> socketChannelType() {
      */
     public static boolean supportsDomainSockets(EventLoopGroup eventLoopGroup) {
         requireNonNull(eventLoopGroup, "eventLoopGroup");
-        final TransportType found = findOrNull(eventLoopGroup);
+        if (!(eventLoopGroup instanceof IoEventLoopGroup)) {
```

Comment:
`IoEventLoopGroup` is now required to get the transport type bound to the event loop. https://netty.io/news/2025/04/03/4-2-0.html#:~:text=with%20virtual%20threads.-,multithreadioeventloopgroup,-In%20previous%20versions